api/v1alpha1/project_config_types.go

```diff
@@ -46,6 +86,29 @@ type PromotionPolicy struct {
 	AutoPromotionEnabled bool `json:"autoPromotionEnabled,omitempty" protobuf:"varint,2,opt,name=autoPromotionEnabled"`
 }
 
+// WebhookReceiverConfig describes the configuration for webhook receivers.
+type WebhookReceiverConfig struct {
+	// GitHub contains the configuration for a webhook receiver that is compatible with
+	// GitHub payloads.
+	GitHub *GitHubWebhookReceiver `json:"github,omitempty" protobuf:"bytes,1,opt,name=github"`
+}
+
+type GitHubWebhookReceiver struct {
+	// Secret contains a reference to a Secret in the same namespace as the ProjectConfig.
```

Comment:
The linter is usually picky about the first word of a field comment being the name of the field itself. (We might have linting disabled for *_type.go files. Don't recall, but we should still honor that convention.)

```suggestion
	// SecretRef contains a reference to a Secret in the same namespace as the ProjectConfig.
```